### PR TITLE
fix: radio preferences don't show description and geocoding fields

### PR DESCRIPTION
**Pull request by KrissDrawing**
# Pull Request Template

## Issue Overview

This PR addresses #3835 and #3830

- [x] This change addresses the issue in full
- [ ] This change addresses only certain aspects of the issue
- [ ] This change is a dependency for another issue
- [ ] This change has a dependency from another issue

## Description

It aligns radio fields behaviour to checkboxes (for multiselect options), so it adds description, and geocoding fields (address, address holder name / relationship).
This component is reused both for programs and preferences, so even that is preferences change it might affect programs (but it shouldn't).

## How Can This Be Tested/Reviewed?

On partners portal set all preferences options to exclusive (for selected preference). Also test with `Can applicants opt out?` set to `yes` for preference. That will work both for digital app and paper app.
- make sure user can't submit empty radio preference for digital app.
- make sure programs, and checkboxes work as before.

## Checklist:

- [x] My code follows the style guidelines of this project
- [ ] I have added QA notes to the issue with applicable URLs
- [x] I have performed a self-review of my own code
- [x] I have reviewed the changes in a desktop view
- [x] I have reviewed the changes in a mobile view
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules
- [ ] I have assigned reviewers
- [ ] I have run `yarn generate:client` and/or created a migration if I made backend changes that require them
- [ ] My commit message(s) is/are polished, and any breaking changes are indicated in the message and are well-described
- [ ] Commits made across packages purposefully have the same commit message/version change, else are separated into different commits

## Reviewer Notes:

Steps to review a PR:

- Read and understand the issue, and ensure the author has added QA notes
- Review the code itself from a style point of view
- Pull the changes down locally and test that the acceptance criteria is met
- Also review the acceptance criteria on the Netlify deploy preview (noting that these do not yet include any backend changes made in the PR)
- Either explicitly ask a clarifying question, request changes, or approve the PR if there are small remaining changes but the PR is otherwise good to go

## On Merge:

If you have one commit and message, squash. If you need each message to be applied, rebase and merge.

### CODE REVIEW

**Review by mcgarrye — APPROVED** [review]
Looks good to me but I am not sure I tested all flows for this. 2nd reviewer should definitely be thorough

File: shared-helpers/src/views/multiselectQuestions.tsx [hunk 1]
```diff
@@ -314,47 +298,121 @@ export const getCheckboxOption = (
...
+      acc[key] = value
+    }
+    return acc
+  }, {})
+
+  // Convert "true" to true
+  cleanedObj = Object.entries(cleanedObj).reduce((acc, [key, value]) => {
+    if (
+      value === "true" &&
+      !(key.includes(AddressHolder.Name) || key.includes(AddressHolder.Relationship))
```
Comment (ludtkemorgan) [inline shared-helpers/src/views/multiselectQuestions.tsx:394]: Why do we need to exclude these two keys? Is it so we don't overwrite somebody who writes "true" into the field?

**Review by ludtkemorgan — APPROVED** [review]
This looks good. Just one question

File: shared-helpers/src/views/multiselectQuestions.tsx (hunk 1 above)
Comment (KrissDrawing) [reply shared-helpers/src/views/multiselectQuestions.tsx:394]: I believe so, not quite sure, but i remember that was the problem, so it's probably it 😄

**Pull request merged**